packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -3995,7 +3995,7 @@ abstract class Element extends DiagnosticableTree implements BuildContext {
   T? findAncestorWidgetOfExactType<T extends Widget>() {
     assert(_debugCheckStateIsActiveForAncestorLookup());
     Element? ancestor = _parent;
-    while (ancestor != null && ancestor.widget.runtimeType != T)
```

Comment:
Doesn't this change the semantics when inheritance comes into play:

```dart
void main() {
  Bar bar = Bar();
  print(bar.runtimeType != Foo);  // true
  print(bar is! Foo);  // false
}

class Foo { }

class Bar extends Foo { }
```

Hence, it wouldn't be an _exact_ type lookup anymore?

Reply:
Oh shoot, you're right!